### ISSUE TITLE
Adding a new layer causes LNK2001 error

### ISSUE THREAD

**Issue by TomC868** [Microsoft/caffe#38]
I followed this guide https://github.com/BVLC/caffe/wiki/Development for adding a new layer, but when I try to build it in Visual Studio, I'll get this error:

error LNK2001: unresolved external symbol "public: __cdecl caffe::NewLayerDataParameter::NewLayerDataParameter(class caffe::NewLayerDataParameter const &)" (??0NewLayerDataParameter@caffe@@QEAA@AEBV01@@Z)	C:\BuildDirectory\caffe-master\windows\pycaffe\new_layer_data_layer.obj

Interestingly, the new layer builds and works fine using cmake on Linux.

**Comment by sasagalic-MSFT** [comment]
Did you add your new cpp files to Visual Studio project files?

**Comment by TomC868** [comment]
Yes, I added the new .cpp files into libcaffe/src/layers and the new .hpp into libcaffe/include/layers using "Add Existing Item".

**Comment by TomC868** [comment]
I played around with it and was able to fix it and get it to build.

NewLayerDataParameter is defined in the caffe.proto file. If I don't instantiate NewLayerDataParameter in the NewLayer.cpp code, then I won't get this linking error and it builds fine.

I believed ProtoCompile.cmd takes caffe.proto and creates caffe.pb.cc and caffe.pb.h with my new parameters. In an act of rage, I deleted both caffe.pb.cc and caffe.pb.h from the directory and had them created again by ProtoCompile.cmd, which after that, the project was able to link fine.

The ProtoCompile.cmd looks like it does a check to see if caffe.pb.h has changed, so it's still a mystery as to why I had to delete the two generated files for the update to take place.
